lending_pool/src/factory.rs

```diff
@@ -7,39 +7,39 @@ pub trait FactoryModule {
     fn create_pool(
         &self,
         base_asset: &TokenIdentifier,
-        lending_pool_address: &Address,
-        r_base: Self::BigUint,
-        r_slope1: Self::BigUint,
-        r_slope2: Self::BigUint,
-        u_optimal: Self::BigUint,
-        reserve_factor: Self::BigUint,
-        bytecode: &BoxedBytes,
-    ) -> Address {
+        lending_pool_address: &ManagedAddress,
+        r_base: BigUint,
+        r_slope1: BigUint,
+        r_slope2: BigUint,
+        u_optimal: BigUint,
+        reserve_factor: BigUint,
+        bytecode: &ManagedBuffer,
+    ) -> ManagedAddress {
         let code_metadata = CodeMetadata::UPGRADEABLE;
-        let amount = Self::BigUint::from(0u32);
+        let amount = self.types().big_uint_zero();
 
-        let mut arg_buffer = ArgBuffer::new();
-        arg_buffer.push_argument_bytes(base_asset.as_esdt_identifier());
-        arg_buffer.push_argument_bytes(lending_pool_address.as_bytes());
-        arg_buffer.push_argument_bytes(r_base.to_bytes_be().as_slice());
-        arg_buffer.push_argument_bytes(r_slope1.to_bytes_be().as_slice());
-        arg_buffer.push_argument_bytes(r_slope2.to_bytes_be().as_slice());
-        arg_buffer.push_argument_bytes(u_optimal.to_bytes_be().as_slice());
-        arg_buffer.push_argument_bytes(reserve_factor.to_bytes_be().as_slice());
+        let mut arg_buffer = ManagedArgBuffer::new_empty(self.type_manager());
+        arg_buffer.push_arg(base_asset);
+        arg_buffer.push_arg(lending_pool_address);
+        arg_buffer.push_arg(r_base);
+        arg_buffer.push_arg(r_slope1);
+        arg_buffer.push_arg(r_slope2);
+        arg_buffer.push_arg(u_optimal);
+        arg_buffer.push_arg(reserve_factor);
```

Comment:
Could use a deploy proxy here instead. Example: 

https://github.com/ElrondNetwork/sc-bridge-elrond/blob/20fb1c63150deec5228dac03bfaa3233f58d86a4/multisig/src/setup.rs#L75-L79

https://github.com/ElrondNetwork/sc-bridge-elrond/blob/main/multisig/src/setup.rs#L363

Reply:
will be refactored in another PR